### ISSUE TITLE
Step 6's re-scan ships no command, and the obvious improvisation mis-reports under zsh without changing any count

### ISSUE THREAD

**Issue by chibuzorokoh** [rebelytics/one-skill-to-rule-them-all#174]
**Target:** `references/weekly-review.md`, Step 6 ("re-scan, then mark ACTIONED"), and the Step 1 reconciliation guard that says to keep the file list.
**Verified at:** upstream `main` (`7518a85`); also present in the v3.1.0 install (`2967fa5`).

## What happened

Step 6 asks the review to compare the current listing of `observation-log/` against the Step 1 file list, and says the check "is one `ls` against a list already in hand". No command is shipped. Shell state does not carry between tool calls in most harnesses, so "a list already in hand" gets rebuilt from whatever the agent improvises.

In a scheduled review on macOS (login shell zsh 5.9), the improvisation kept the Step 1 names in a space-separated variable and looped `for f in $now`. zsh does not word-split an unquoted parameter expansion, so each loop body ran once over the whole string, and the output listed every file as both arrived and gone. The count guard (N files = N files) passed throughout, because it counts files, not names.

That output contradicted itself and was caught. A quieter variant would not be: a membership test that is always false reads as "every file arrived", and the reverse test reads as "none arrived", which is exactly the delta the step exists to catch.

The same class turned up twice more in later runs on the same machine:

- A bare `echo ====` in a compound command aborted it with `zsh:1: === not found` (zsh's `=word` expansion looks up a command called `===`), and everything after it was lost.
- An unrelated set comparison written as `printf '%s\n' $list | …` produced a one-element "set" and a nonsense comparison. The instrument printed a size of 1 where 48 were expected.

## Reproduction

```bash
for sh in zsh bash; do
  echo "## $sh"
  $sh -c 'names="a.md b.md c.md"; i=0; for f in $names; do i=$((i+1)); done; echo "iterations=$i (3 expected)"'
  $sh -c 'echo ==== ; echo "after (reached)"' 2>&1
done
```

| Shell | Unquoted `$names` loop | `echo ====; echo after` |
|---|---|---|
| zsh 5.9 | `iterations=1` | `zsh:1: === not found`, and the second command never runs |
| GNU bash 3.2.57 (macOS `/bin/bash`) | `iterations=3` | `====` then `after (reached)` |

## Why the current text did not prevent it

- **The step ships prose where its neighbours ship commands.** The scan and the id derivation both come as guarded snippets. The re-scan, whose output is a set difference, comes as a sentence. #101 was the same shape for archival, and the fix there was to put the command next to the rule.
- **The count guard cannot see this failure.** A broken comparison can invent or hide a delta without changing any count. The existing guard is about the number of files, and the defect lives in the names.
- **The portability notes cover one zsh difference, not the class.** They cover unmatched globs (and `\<`), but not that unquoted `$var` is not word-split, or that a word starting with `=` is expanded as a command path. Both break commands that run correctly under bash.

## Suggested fix

1. **Ship the re-scan as a line-oriented set difference, with the Step 1 list kept on disk.** Tested under zsh 5.9 and bash 3.2.57, with a planted file created after Step 1 reported as `arrived` in both:

   ```bash
   # Step 1 — write the list to a file, never hold it in a variable
   d="[ABSOLUTE PATH]/skill-observations/observation-log"
   l="[ABSOLUTE PATH]/skill-observations/review-step1-list.txt"
   find "$d" -maxdepth 1 -name '*.md' -exec basename {} \; | LC_ALL=C sort > "$l"

   # Step 6 — re-derive both paths in this call; compare names, not counts
   d="[ABSOLUTE PATH]/skill-observations/observation-log"
   l="[ABSOLUTE PATH]/skill-observations/review-step1-list.txt"
   now="[ABSOLUTE PATH]/skill-observations/review-step6-list.txt"
   find "$d" -maxdepth 1 -name '*.md' -exec basename {} \; | LC_ALL=C sort > "$now"
   n_now=$(find "[ABSOLUTE PATH]/skill-observations/observation-log" -maxdepth 1 -name '*.md' | wc -l | tr -d ' ')
   n_list=$(wc -l < "$now" | tr -d ' ')
   [ "$n_now" -eq "$n_list" ] || { echo "RESCAN COMMAND BROKEN — $n_now files, $n_list listed"; exit 1; }
   echo "arrived:"; LC_ALL=C comm -13 "$l" "$now"
   echo "gone:";    LC_ALL=C comm -23 "$l" "$now"
   ```

   A positive control is one extra line: create a throwaway `.md` in a copy of the directory after Step 1, and confirm `comm -13` names it.

2. **Portability notes, two sentences:** under zsh, an unquoted `$var` is not split into words, so never carry a list in a variable and loop over it (use a file and `while IFS= read -r`, or `comm`); and a word beginning with `=` is expanded as a command path, so quote it (`echo '===='`).

## Principle

A step that ships no command is implemented by whatever the agent improvises, in whatever shell it runs, and the guards that protect the shipped snippets do not reach it. Where a step's output is a set difference, ship the difference as a line-oriented command with a planted-member control. The failure it has to catch is a delta, and a broken comparison can invent or hide a delta without moving any count.

**Comment by rebelytics** [comment]
Thanks for this one. It's in the queue and will make it into one of the next weekly releases.
